Automattic/woocommerce-payments
# Prevent challenging Klarna inquiries

Fixes #9703 

#### Changes proposed in this Pull Request

Evidence submitted for Klarna inquiries are ignored by Stripe. This PR removes the ability to submit evidence for Klarna inquiries.

Changes for Klarna Inquiries:
 - "Submit Evidence" button hidden
 - Steps to respond updated to remove reference to "Submit Evidence" 
 - "Email the customer" mailto link updated to remove references to disputes in email body.


It's worth noting that submitting evidence is not blocked, the merchant can still manually navigate to `/wp-admin/admin.php?page=wc-admin&path=%2Fpayments%2Fdisputes%2Fchallenge&id=<dispute_id>` to challenge the inquiry. While it would be possible to prevent this I opted for the simple approach. If we decided to prevent access to this page, API calls should also be blocked by the same logic. 


**Before**

![image](https://github.com/user-attachments/assets/beffcc75-c6a0-4a05-92be-4cfaeb4f2642)

**After**

![image](https://github.com/user-attachments/assets/0b974b5b-3412-4b3d-a6f4-99aad6fa46ad)


**Email Copy Changes**
```diff
Hello <name>,

-We noticed that on <inquiry date>, you disputed a <amount> charge on <purchase date>. We wanted to contact you to make sure everything was all right with your purchase and see if there's anything else we can do to resolve any problems you might have had.
+We noticed that on <inquiry date>, you raised a question with your payment provider about a <amount> charge made on <purchase date>. We wanted to reach out to ensure everything is all right with your purchase and to see if there’s anything we can do to resolve any problems you might have had.

-Alternatively, if the dispute was a mistake, you can easily withdraw it by calling the number on the back of your card. Thank you so much - we appreciate your business and look forward to working with you.
+Alternatively, if this was a mistake please contact your payment provider to resolve it. Thank you so much - we appreciate your business and look forward to working with you.
```

**Email Copy**

![Screenshot 2024-11-18 at 5 13 50 PM](https://github.com/user-attachments/assets/3e054481-d871-42bf-9bc7-3f256b61c148)




#### Testing instructions

1. Enable Klarna as an alternate payment method in WooPayments
2. Purchase a cart using Klarna with the customer email address `customer+disputed-return@email.us`
3. Visit `WooPayments > Disputes` and click on the newly created Klarna Inquiry (this may take a few minutes to show up)
4. Confirm `Submit Evidence` does not show up.
5. Click "Email the customer" to confirm the mailto copy doesn't mention disputes.


#### Other Considerations

The `DisputeNotice` component has not been updated. There is a bit of a contradiction between the notice and the docs

![image](https://github.com/user-attachments/assets/5a44e0af-1b68-46e9-8d1e-0b9d00ef36b7)


> You can challenge their claim if you believe it’s invalid. Not responding will result in an automatic loss - `DisputeNotice`

>  Card issuers can convert inquiries to disputes at their discretion - [WooPayments Docs](https://woocommerce.com/document/woopayments/fraud-and-disputes/managing-disputes/#inquiries:~:text=Card%20issuers%20can%20convert%20inquiries%20to%20disputes%20at%20their%20discretion)

-------------------


**Post merge**
